### PR TITLE
Update formating to pass pre-commits checks

### PR DESCRIPTION
**Pull request by Czaki**
In this PR I reformat cookiecutter template for plugins to pass* pre-commit run. 

The current output of the cookiecutter fails pre-commit checks. 

Especially `isort` removes all import from `__init__` file because of lack of `__all__` entry. 

Code from this PR pass almost tall pre-commit checks. The only exception is `setup-cfg-fmt` because it will remove comments that I think may be important for a new user.

### CODE REVIEW

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -69,6 +69,21 @@ def validate_manifest():
...
+        pass
+{% if cookiecutter.install_precommit == 'y' %}
+    # try to install and update pre-commit
+    try:
+        print("install pre-commit ...")
+        subprocess.run(["pip", "install", "pre-commit"], stdout=subprocess.DEVNULL)
+        print("updating pre-commit...")
+        subprocess.run(["pre-commit", "autoupdate"], stdout=subprocess.DEVNULL)
+        subprocess.run(["git", "add", "."])
+        subprocess.run(["pre-commit", "run", "black", "-a"], capture_output=True)
```
Comment (Czaki) [inline hooks/post_gen_project.py:82]: fix formating, dependent on line lengths

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -150,18 +175,4 @@ def validate_manifest():
    https://github.com/chanzuckerberg/napari-hub/blob/main/docs/customizing-plugin-listing.md
 """
 
-{% if cookiecutter.github_repository_url != 'provide later' %}
```
Comment (Czaki) [inline hooks/post_gen_project.py:153]: I do not know why there is a check for `'provide later'`. I remove it, but maybe I do not know the scenario where it is needed.

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_widget.py [hunk 3]
```diff
@@ -6,9 +6,14 @@
...
 Replace code below according to your needs.
 """
+from typing import TYPE_CHECKING
+
 from magicgui import magic_factory
 from qtpy.QtWidgets import QHBoxLayout, QPushButton, QWidget
 
+if TYPE_CHECKING:
+    import napari
+
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_widget.py:16]: to pass flake8 check.

File: hooks/post_gen_project.py (hunk 2 above)
Comment (Czaki) [reply hooks/post_gen_project.py:153]: but it looks like #75 was wrong if introduced, and proper was introduced in #78? 
I still do not see why it depends on providing the repository.

Comment (Czaki) [reply hooks/post_gen_project.py:153]: So it looks like in this moment it is enough to check if user ask for pre-commit

**Review by nclack — APPROVED** [review]
looks good!  thanks

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**